scripts/lib/CIME/XML/compsets.py

```diff
@@ -45,33 +44,36 @@ def get_compset_var_settings(self, compset, grid):
         Variables can be set in config_compsets.xml in entry id settings with compset and grid attributes
         find and return id value pairs here
         '''
-        nodes = self.get_nodes("entry")
-        # Get an empty entryid obj to use
-        entryidobj = EntryID()
+        entries = self.get_optional_child("entries")
```

Comment:
We can avoid the recursive search by grabbing the "entries" element and using that to find "entry" elements.